assembl/static2/js/app/components/harvesting/taxonomyOverflowMenu.jsx

```diff
@@ -1,54 +1,113 @@
 // @flow
 import React from 'react';
-import { Popover } from 'react-bootstrap';
+import { Popover, Button } from 'react-bootstrap';
 import { Translate } from 'react-redux-i18n';
 import classnames from 'classnames';
 import { extractNatures, extractActions, NatureIcons, ActionIcons } from '../../utils/extractQualifier';
 
-const TaxonomyOverflowMenu = (handleClick: Function, extractNature: ?string, extractAction: ?string) => (
-  <Popover id="taxonomy" className="taxonomy-menu overflow-menu">
-    <div className="pointer taxonomy-label taxonomy-label-border">
-      <Translate value="harvesting.move" />
-    </div>
-    <div className="taxonomy-category">
-      <Translate value="harvesting.qualifyNature" />
-    </div>
-    <div className="pointer">
-      {extractNatures.map(n => (
-        <div
-          onClick={() => {
-            handleClick('nature', n.qualifier);
-          }}
-          key={n.qualifier}
-          className={classnames('taxonomy-label', { active: extractNature === n.qualifier })}
-        >
-          <NatureIcons qualifier={n.qualifier} />
-          <div className="nature-label">
-            <Translate value={n.label} />
+type TaxonomyOverflowMenuProps = {
+  handleClick: Function,
+  extractNature?: string,
+  extractAction?: string,
+  onCloseClick: Function,
+  top: number
+}
+
+type taxonomyOverflowMenuState = {
+  nature: ?string,
+  action: ?string
+}
+
+class TaxonomyOverflowMenu extends React.Component<*, TaxonomyOverflowMenuProps, taxonomyOverflowMenuState> {
```

Comment:
avoid `*` for default props (either it is `void` or it is known (here I guess : `{ extractNature: null, extractAction: null }` ))